### ISSUE TITLE
snapclient crashes when soundcard is in use

### ISSUE THREAD

**Issue by quantenschaum** [badaix/snapcast#121]
There are several things with the client and a blocked soundcard. (bocked = in use by other program)
1. When starting the client and the soundcard is blocked, it waits for it to become available (good)
2. When started with an available soundcard and starting playback when the card got blocked in the meantime, it crashes (bad)

It would be nice, if the client would access the card only when needed (when playback starts) and releases it afterwards. If the card is blocked, it should wait for it, but not crash. This way the client could share the soundcard with other applications.

**Comment by badaix** [comment]
Are you talking about Linux/Alsa?
I cannot reproduce this. I can start lots of snapclients, aplay or whatever audio players. All are playing in parallel as intended.
What is your setup (OS, soundcard, ...)? How to reproduce this?

**Comment by quantenschaum** [comment]
I am using Ubuntu Xenial/Trusty and Minibian on the Pi. The soundcard passed to snapclient is a `hw` device, not `dmux`, so parallel playback is not possible. The clients may start, but then try point 2 (play with snapclient, stop the playback, play with other app to `hw` device, then try to playback with snapclient).

**Comment by unimatrix27** [comment]
I made the same experience with snapclient crashing sometimes but not always.When I then used the plughw device, it did not crash any more. I dont understand ALSA well enough to have any idea about how this really makes a difference. I am using snapclient on a machine that is always running KODI (OSMC) and is used for my living room TV. I am surprised that it actually plays in parallel, so the sound is mixed together. I had not expected that at all. Even when KODI is playing PCM 5.1 sound (not passthrough but decoded) I can still play with snapclient .
